src/common/opencl.c

```diff
@@ -1399,33 +1399,37 @@ int dt_opencl_lock_device(const int pipetype)
   int *priority = (int *)malloc(prio_size);
   int mandatory;
 
-  switch(pipetype)
-  {
-    case DT_DEV_PIXELPIPE_FULL:
+    if((pipetype & DT_DEV_PIXELPIPE_FULL) == DT_DEV_PIXELPIPE_FULL)
```

Comment:
These types of bitbucket checks are usually be simplified to just
`if(pipetype & DT_DEV_PIXELPIPE_FULL)`
same with others below.

Unless I'm mistaken? I see this pattern repeated everywhere in the code now (at least 93 times for just DT_DEV_PIXELPIPE according to grep). Is there a reason for this somehow in this case? Or just a copy/paste bonanza. Maybe a nice cleanup post release?

Anyway, good catch/bug hunt!

Reply:
Copy paste from other code doing the same thing. I'm happy to change it if you think it's better

Reply:
It looks like you could have just masked off the FAST bit in the switch:
`switch(pipetype & ~DT_DEV_PIXELPIPE_FAST)`

Reply:
too... many... options... ;) 

@TurboGit what would you like me to do here

Reply:
I suppose I would go for : `switch(pipe_type & DT_DEV_PIXELPIPE_ANY) `

This will require less modifications after and will make the code easier to read.

Reply:
ok will do

Reply:
done